src/Prefetch.cpp

```diff
@@ -205,11 +205,24 @@ class InjectPrefetch : public IRMutator {
                 // either read or written.
                 for (const auto &b : boxes_read) {
                     const string &buf_name = b.first;
+                    Box prefetch_box = b.second;
 
                     // Only prefetch the region that is in bounds.
                     Box bounds = buffer_bounds(buf_name, b.second.size());
-                    Box prefetch_box = box_intersection(b.second, bounds);
 
+#if 1
+                    // TODO: Consider enabling this, which adds likely intrinsics to the
+                    // box prior to clamping it to the bounds of the buffer. This leads
+                    // the loops to be split up such that the steady state does not have
+                    // any boundary condition logic. On matrix multiplication on Hexagon,
+                    // this reduced the runtime by ~4%.
+                    for (auto &i : prefetch_box.bounds) {
+                        i.min = likely(i.min);
```

Comment:
Hmm, I think this *should* work, but something is going on in LLVM that doesn't quite make sense. Either way, this change (likely vs. likely_if_innermost vs. no likely at all) is barely affecting the runtime, so I think I'm going to just remove this code for now, we can look into it again it makes sense and makes a difference on another workload. FWIW, this inner loop doesn't have much work (only ~20 VLIW packets), so I would expect the overhead of the boundary logic to be significant in this workload (if it is anywhere), and that doesn't seem to be the case.